playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

```diff
@@ -941,11 +943,23 @@ public void onTimelineChanged(@NonNull Timeline timeline, int reason) {
             }
         }
 
-        if (reason == Player.TIMELINE_CHANGE_REASON_SOURCE_UPDATE && getDuration() != TIME_UNSET) {
-            if (!isLoadedMetaDataFired)  {
-                sendDistinctEvent(PlayerEvent.Type.LOADED_METADATA);
+        if (reason == Player.TIMELINE_CHANGE_REASON_SOURCE_UPDATE) {
+            if (getDuration() != TIME_UNSET) {
+                if (!isLoadedMetaDataFired) {
+                    sendDistinctEvent(PlayerEvent.Type.LOADED_METADATA);
+                }
+                sendDistinctEvent(PlayerEvent.Type.DURATION_CHANGE);
+            }
+
+            if (player.getCurrentManifest() instanceof DashManifest) {
+                if (((DashManifest) player.getCurrentManifest()).getPeriodCount() > 0) {
+                    List<EventStream> eventStreamsList = ((DashManifest) player.getCurrentManifest()).getPeriod(0).eventStreams;
+                    if (!eventStreamsList.isEmpty()) {
```

Comment:
eventStreamsList check not null

Reply:
It is non null internally. Exo always has an instance.